rts/Sim/Weapons/Weapon.cpp

```diff
@@ -256,6 +256,13 @@ void CWeapon::UpdateWeaponPieces(const bool updateAimFrom)
 
 void CWeapon::UpdateWeaponVectors()
 {
+	ZoneScoped;
+
+	float3 newErrorVector = (errorVector + errorVectorAdd);
+	if (newErrorVector.SqLength() <= 1.0f)
+		errorVector = newErrorVector;
```

Comment:
Why did you move this into `UpdateWeaponVectors()` from `Update()` ?

Reply:
This code smells a bit, but I don't think it's a correct place to be as `w->UpdateWeaponVectors();` is called from different places not only from `unit->UpdateWeaponVectors();`